hazelcast/src/main/java/com/hazelcast/security/permission/SqlPermission.java

```diff
@@ -63,6 +70,8 @@ protected int initMask(String[] actions) {
                     mask |= CREATE_DATACONNECTION;
                 } else if (ActionConstants.ACTION_DROP_DATACONNECTION.equals(action)) {
                     mask |= DROP_DATACONNECTION;
+                } else if (ActionConstants.ACTION_READ.equals(action)) {
```

Comment:
I am afraid that this might be misleading. Usually `read` permission means "read contents" not "read definition", but for MAPPING this is tricky. Currently we do not have content-permissions on MAPPING level, only on IMap level.

Reply:
That's why I put a note above; also I prefer to be consistent and align the view permission with the current style for mappings.

Reply:
consistency is good and comment is for us. but I meant that it will be misleading for the users and hz admins. Having `READ` permission for `MAPPING` does not allow to select data from the mapping. Of course we can document it (but who reads docs?).

Reply:
So, I assume, you propose to rework mapping permissions with independent action names?

Reply:
not all, only use dedicated action name for viewing mapping definition, as stated in the the PR description. Would it be possible to add `create-mapping` and `drop-mapping` actions in backward- and RU-compatible way? Current state where `SqlPermission` applies to mapping unless you use special actions is not obvious for the user.

Reply:
https://github.com/hazelcast/hazelcast/pull/25662/commits/99146d9195d5b24385d94ae77d72e4d6421c041f

Reply:
> Would it be possible to add create-mapping and drop-mapping actions in backward- and RU-compatible way?

I think, yes.

Reply:
HZ-3462